### PR TITLE
VBLOCKS-2030 | API stubs and error checking

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Pull Request Details

https://issues.corp.twilio.com/browse/VBLOCKS-2030

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Squashed erroneous commits if necessary
* [ ] Re-tested if necessary

### CODE REVIEW

File: lib/twilio/audiohelper.ts [hunk 1]
```diff
@@ -342,6 +348,35 @@ class AudioHelper extends EventEmitter {
     });
   }
 
+  /**
+   * Adds an {@link AudioProcessor}. If an {@link AudioProcessor} exists in the
```
Comment (mhuynh5757) [inline lib/twilio/audiohelper.ts:352]: In my opinion, the wording here seems to imply "If an AudioProcess (already) exists in the..." due to grammatical structure. My gut instinct reading this is that the functionality only works on the second or further invocation of the function, which I am assuming is not true. I'm trying to think of how I would word this... maybe something like

```
Adds an AudioProcessor.

Once the AudioProcessor contains at least one AudioProcessor, the AudioHelper routes the input audio stream to the AudioProcessor.
```

Comment (mhuynh5757) [reply lib/twilio/audiohelper.ts:352]: I understand that currently we do not support multiple AudioProcessors, so maybe you'd want to reword that.

File: lib/twilio/audiohelper.ts [hunk 2]
```diff
@@ -342,6 +348,35 @@ class AudioHelper extends EventEmitter {
...
+   *
+   * Only one {@link AudioProcessor} can be added at this time.
+   * @param processor
+   */
+  addProcessor(processor: AudioProcessor): void {
+    if (this._processor) {
+      throw new NotSupportedError('Adding multiple AudioProcessors is not supported at this time.');
+    }
+
+    if (!processor) {
```
Comment (mhuynh5757) [inline lib/twilio/audiohelper.ts:368]: Do we want to do something a little stricter here like `typeof processor != 'object'`?

File: lib/twilio/audiohelper.ts [hunk 3]
```diff
@@ -342,6 +348,35 @@ class AudioHelper extends EventEmitter {
...
+      throw new InvalidArgumentError('Missing AudioProcessor argument.');
+    }
+
+    if (typeof processor.createProcessedStream !== 'function' ||
+    typeof processor.destroyProcessedStream !== 'function') {
+      throw new InvalidArgumentError('Missing createProcessedStream or destroyProcessedStream.');
+    }
```
Comment (mhuynh5757) [inline lib/twilio/audiohelper.ts:375]:
```suggestion
    if (
      typeof processor.createProcessedStream !== 'function' ||
      typeof processor.destroyProcessedStream !== 'function'
    ) {
      throw new InvalidArgumentError('Missing createProcessedStream or destroyProcessedStream.');
    }
```
Nitpick

**Review by mhuynh5757 — COMMENTED** [review]
Left a few comments, looks good!

File: lib/twilio/audiohelper.ts (hunk 2 above)
Comment (manjeshbhargav) [inline lib/twilio/audiohelper.ts:368]: This change will make sure a value of `null` will not trigger this error:
```suggestion
    if (typeof processor === 'undefined') {
```

File: lib/twilio/audiohelper.ts [hunk 4]
```diff
@@ -342,6 +348,35 @@ class AudioHelper extends EventEmitter {
...
+      throw new NotSupportedError('Adding multiple AudioProcessors is not supported at this time.');
+    }
+
+    if (!processor) {
+      throw new InvalidArgumentError('Missing AudioProcessor argument.');
+    }
+
+    if (typeof processor.createProcessedStream !== 'function' ||
+    typeof processor.destroyProcessedStream !== 'function') {
+      throw new InvalidArgumentError('Missing createProcessedStream or destroyProcessedStream.');
```
Comment (manjeshbhargav) [inline lib/twilio/audiohelper.ts:374]:
```suggestion
    if (typeof processor.createProcessedStream !== 'function') {
      throw new InvalidArgumentError('Missing createProcessedStream() method.');
    }

    if (typeof processor.destroyProcessedStream !== 'function') {
      throw new InvalidArgumentError('Missing destroyProcessedStream() method.');
```

**Review by manjeshbhargav — APPROVED** [review]

File: lib/twilio/audiohelper.ts (hunk 1 above)
Comment (charliesantos) [reply lib/twilio/audiohelper.ts:352]: I'm not sure what you mean by "the functionality only works on the second or further invocation of the function". This was reviewed with the whole team. And another one with you in the IDL. I'm just wondering why the confusion now? Can you elaborate?

File: lib/twilio/audiohelper.ts (hunk 2 above)
Comment (charliesantos) [reply lib/twilio/audiohelper.ts:368]: Good point. But @manjeshbhargav 's suggestion seems better. It accounts for null.

Comment (charliesantos) [reply lib/twilio/audiohelper.ts:368]: Testing it more, your suggestion is better but with a check on null. Will update.

Comment (charliesantos) [reply lib/twilio/audiohelper.ts:368]: We don't want the user to set this to null using this method. I'll check for object and null instead.

**Review by mhuynh5757 — APPROVED** [review]

**Pull request merged**